tools/discovery/discovery.go

```diff
@@ -72,7 +72,13 @@ func refEnginesFetching(parsedName map[string]string) (*object.RefEngines, error
 
 	client := &http.Client{Transport: defaultTrans}
 
-	resp, err := client.Get(u.String())
+	req, err := http.NewRequest("GET", u.String(), nil)
```

Comment:
I think `NewRequest` isintended as a convenience function for folks POSTing data and such.  In #23, I've [instantiated a `Request` structure manually][1], and that's about as compact and works in my command-line testing (although I still don't have Go mocks for fake network access).

[1]: https://github.com/xiekeyang/oci-discovery/pull/23/files#diff-855940c550098b12c1e2721e8921c216R85